### PR TITLE
chore(contentful-apps): Add Contentful apps to monorepo

### PR DESCRIPTION
**Pull request by RunarVestmann**
# Add Contentful apps to monorepo

## What

* I've been adding a couple of Contentful apps to extend the functionality inside the Contentful content management system and would like source control the apps and also provide a simple way for other devs to add Contentful apps in the future.
* The solution as discussed here: https://islandis.slack.com/archives/C019VN1KG04/p1676973267211549 was to create a Next.js application inside the monorepo that would host our Contentful Apps.

## Why

* We want to source control all code related to the Ísland.is project and that includes Contentful apps.

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

**Comment by RunarVestmann** [comment]
Building the Next.js app results in the following TypeError: 
image

File: apps/contentful-apps/infra/contentful-apps.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    .image('contentful-apps')
+    .namespace('contentful-apps')
+    .serviceAccount('contentful-apps')
+    .ingress({
+      primary: {
+        host: {
+          dev: 'contentful-apps',
+          staging: 'contentful-apps',
+          prod: 'contentful-apps.devland.is',
+        },
```
Comment (eirikurn) [inline apps/contentful-apps/infra/contentful-apps.ts:14]: Note @island-is/devops. This app is to expand the UI in our CMS for developers and content editors. We don't need dev/staging/prod environments and we'd generally like to release updates outside the normal release trains.

So we'll probably deploy this like our existing "contentful-extensions-translations" project. Only host it on dev with CD, without cognito (nothing sensitive here) and point cognito to `https://contentful-apps.dev01.devland.is/something`.

My only concern is downtime during deployment. The existing project is static but this one is nextjs. Not sure if that's worse or if it's a big problem at all.

 ideally we can have zero-downtime continuous delivery for this app. It's used by developers and content editors

**Review by eirikurn — APPROVED** [review]
Love it!

File: apps/contentful-apps/project.json [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        "passWithNoTests": true
+      }
+    },
+    "lint": {
+      "executor": "@nrwl/linter:eslint",
+      "outputs": ["{options.outputFile}"],
+      "options": {
+        "lintFilePatterns": ["apps/contentful-apps/**/*.{ts,tsx,js,jsx}"]
+      }
+    }
```
[the remark this answers is not in the document]
Comment (RunarVestmann) [reply apps/contentful-apps/project.json]: Done 👍

File: apps/contentful-apps/project.json [hunk 3]
```diff
@@ -0,0 +1,54 @@
+{
+  "root": "apps/contentful-apps",
+  "sourceRoot": "apps/contentful-apps",
+  "projectType": "application",
+  "targets": {
+    "build": {
```
[the remark this answers is not in the document]
Comment (RunarVestmann) [reply apps/contentful-apps/project.json:6]: Done 👍

File: apps/contentful-apps/infra/contentful-apps.ts (hunk 1 above)
Comment (busla) [reply apps/contentful-apps/infra/contentful-apps.ts:14]: > Note @island-is/devops. This app is to expand the UI in our CMS for developers and content editors. We don't need dev/staging/prod environments and we'd generally like to release updates outside the normal release trains.
> 
> So we'll probably deploy this like our existing "contentful-extensions-translations" project. Only host it on dev with CD, without cognito (nothing sensitive here) and point cognito to `https://contentful-apps.dev01.devland.is/something`.
> 
> My only concern is downtime during deployment. The existing project is static but this one is nextjs. Not sure if that's worse or if it's a big problem at all.
> 
> ideally we can have zero-downtime continuous delivery for this app. It's used by developers and content editors



regarding the "downtime during" deployment issue, are you referring to the indexer issue?

Comment (busla) [reply apps/contentful-apps/infra/contentful-apps.ts:14]: We can't point cognito to that url, but we can disable authentication on the ingress for this service

Comment (RunarVestmann) [reply apps/contentful-apps/infra/contentful-apps.ts:14]: I finally got the TypeScript stuff to work. Can someone from @island-is/devops review this PR?

**Review by busla — APPROVED** [review]
LGTM

**Review by oddsson — APPROVED** [review]

File: apps/contentful-apps/proxy.config.json [hunk 4]
```diff
@@ -0,0 +1,6 @@
+{
+  "/api": {
+    "target": "http://localhost:4444",
+    "secure": false
+  }
+}
```
Comment (eirikurn) [inline apps/contentful-apps/proxy.config.json:6]: This app doesn't need to proxy any api stuff, right? Can remove this and from the configuration/server.ts

**Review by eirikurn — APPROVED** [review]
Great job upgrading typescript! 🎉

File: apps/contentful-apps/proxy.config.json (hunk 4 above)
Comment (RunarVestmann) [reply apps/contentful-apps/proxy.config.json]: Ok, I've removed the proxy.config.json and also taken it out of the server.ts file.

**Review by diljath — APPROVED** [review]

**Comment by eirikurn** [comment]
Split this PR apart from the TypeScript update which is now here: #10269 - Took some time to upgrade types in a project and want to upgrade to 4.9 since we're doing this.

The TypeScript PR probably needs to be merged before this PR can be merged.

Thanks for your effort in these two PRs! 👍

**Comment by busla** [comment]
Nice!

This will help with both mobile-app integration and NX upgrade!

Great job 🙌

**Review by busla — APPROVED** [review]
LGTM

**Pull request merged**